hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azure/ITestWasbUriAndConfiguration.java

```diff
@@ -640,4 +640,28 @@ public void testUserAgentConfig() throws Exception {
       FileSystem.closeAll();
     }
   }
+
+  @Test
+  public void testCanonicalServiceName() throws Exception {
+    AzureBlobStorageTestAccount testAccount = AzureBlobStorageTestAccount.createMock();
+    Configuration conf = testAccount.getFileSystem().getConf();
+    String authority = testAccount.getFileSystem().getUri().getAuthority();
+    URI defaultUri = new URI("wasbs", authority, null, null, null);
+    conf.set(FS_DEFAULT_NAME_KEY, defaultUri.toString());
+
+    final FileSystem fs0 =  FileSystem.get(conf);
+    // Default getCanonicalServiceName() will try to resolve the host to IP,
+    // because the mock container does not exist, this call is expected to fail.
+    intercept(IllegalArgumentException.class,
+            "java.net.UnknownHostException",
+            ()-> {
+              fs0.getCanonicalServiceName();
+            });
+
+    // clear fs cache
+    FileSystem.closeAll();
```

Comment:
you don't need to do that , just call `FileSystem.newInstance(defaultURI, conf)`, remembering to close the new FS afterwards